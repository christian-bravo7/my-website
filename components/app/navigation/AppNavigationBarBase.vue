<template>
  <nav
    class="fixed top-0 left-0 right-0 z-20 | p-4 md:py-3 md:px-8 | bg-transparent md:bg-gray-50 md:dark:bg-blue-800 | transition-all duration-150"
  >
    <div class="navbar__content">
      <div class="flex items-center">
        <slot name="left" />
      </div>
      <div class="flex items-center">
        <slot name="center" />
      </div>
      <div class="flex items-center">
        <slot name="right" />
      </div>
    </div>
  </nav>
</template>

<script lang="ts">
import { Vue, Component } from 'nuxt-property-decorator';

@Component
export default class AppNavigationBarBase extends Vue { }
</script>

<style lang="scss" scoped>
.navbar {
  &__content {
    @apply grid;

    grid-template-columns: auto 1fr auto;
  }
}
</style>
