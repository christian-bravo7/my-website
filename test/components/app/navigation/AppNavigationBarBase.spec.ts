import { mount } from '@vue/test-utils';

import AppNavigationBarBase from '@/components/app/navigation/AppNavigationBarBase.vue';

describe('AppNavigationBarBase', () => {
  it('should render correclty', () => {
    const wrapper = mount(AppNavigationBarBase);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render content into left slot', () => {
    const wrapper = mount(AppNavigationBarBase, {
      slots: {
        left: 'hello left slot',
      },
    });

    const navbarContent = wrapper.findAll('.navbar__content div');

    const leftSection = navbarContent.at(0);
    const centerSection = navbarContent.at(1);
    const rightSection = navbarContent.at(2);

    expect(leftSection.text()).toBe('hello left slot');
    expect(centerSection.element).toBeEmptyDOMElement();
    expect(rightSection.element).toBeEmptyDOMElement();
  });

  it('should render content into center slot', () => {
    const wrapper = mount(AppNavigationBarBase, {
      slots: {
        center: 'hello center slot',
      },
    });

    const navbarContent = wrapper.findAll('.navbar__content div');

    const leftSection = navbarContent.at(0);
    const centerSection = navbarContent.at(1);
    const rightSection = navbarContent.at(2);

    expect(leftSection.element).toBeEmptyDOMElement();
    expect(centerSection.text()).toBe('hello center slot');
    expect(rightSection.element).toBeEmptyDOMElement();
  });

  it('should render content into right slot', () => {
    const wrapper = mount(AppNavigationBarBase, {
      slots: {
        right: 'hello right slot',
      },
    });

    const navbarContent = wrapper.findAll('.navbar__content div');

    const leftSection = navbarContent.at(0);
    const centerSection = navbarContent.at(1);
    const rightSection = navbarContent.at(2);

    expect(leftSection.element).toBeEmptyDOMElement();
    expect(centerSection.element).toBeEmptyDOMElement();
    expect(rightSection.text()).toBe('hello right slot');
  });

  it('should not render content in slots', () => {
    const wrapper = mount(AppNavigationBarBase);

    const navbarContent = wrapper.findAll('.navbar__content div');

    const leftSection = navbarContent.at(0);
    const centerSection = navbarContent.at(1);
    const rightSection = navbarContent.at(2);

    expect(leftSection.element).toBeEmptyDOMElement();
    expect(centerSection.element).toBeEmptyDOMElement();
    expect(rightSection.element).toBeEmptyDOMElement();
  });

  it('should render content in slots', () => {
    const wrapper = mount(AppNavigationBarBase, {
      slots: {
        left: 'left content',
        center: 'center content',
        right: 'right content',
      },
    });

    const navbarContent = wrapper.findAll('.navbar__content div');

    const leftSection = navbarContent.at(0);
    const centerSection = navbarContent.at(1);
    const rightSection = navbarContent.at(2);

    expect(leftSection.element).not.toBeEmptyDOMElement();
    expect(centerSection.element).not.toBeEmptyDOMElement();
    expect(rightSection.element).not.toBeEmptyDOMElement();
  });
});
