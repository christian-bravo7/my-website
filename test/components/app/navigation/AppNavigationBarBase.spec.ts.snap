// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNavigationBarBase should render correclty 1`] = `
<nav class="fixed top-0 left-0 right-0 z-20 | p-4 md:py-8 md:px-10 | bg-transparent md:bg-gray-50 md:dark:bg-dark-theme-800 | transition-all duration-150">
  <div class="navbar__content">
    <div class="flex items-center"></div>
    <div class="flex items-center"></div>
    <div class="flex items-center"></div>
  </div>
</nav>
`;
